src/qforte/sq_op_pool.cc

```diff
@@ -29,28 +30,42 @@ const std::vector<std::pair< std::complex<double>, SQOperator>>& SQOpPool::terms
 }
 
 void SQOpPool::set_orb_spaces(const std::vector<int>& ref, const std::vector<size_t>& orb_irreps_to_int){
-    int norb = ref.size();
-    if (norb%2 == 0){
-        norb = static_cast<int>(norb/2);
-    } else {
-        throw std::invalid_argument("QForte does not yet support systems with an odd number of spin orbitals.");
+    // compute integer representing reference determiant
+    ref_int_ = 0;
+    int multiplier = 1;
+    for (const auto& occupancy : ref) {
+        ref_int_ += occupancy * multiplier;
+        multiplier = multiplier << 1;
     }
 
-    nocc_ = 0;
-    for (const auto& occupancy : ref){
-        nocc_ += occupancy;
+    // set orbital spaces
+    n_spinorb_ = ref.size();
+    if (n_spinorb_%2 != 0) {
+        throw std::invalid_argument("The total number of spinorbitals must be even!");
     }
 
-    if (nocc_%2 == 0){
-        nocc_ = static_cast<int>(nocc_/2);
-    } else {
-        throw std::invalid_argument("QForte does not yet support systems with an odd number of occupied spin orbitals.");
-    }
+    n_occ_alpha_ = 0;
+    n_occ_beta_ = 0;
+    n_vir_alpha_ = 0;
+    n_vir_beta_ = 0;
 
-    nvir_ = static_cast<int>(norb - nocc_);
+    int is_alpha = true;
```

Comment:
Can we make `is_alpha` be a _bool_, not an int? The `is_alpha` check below could just be `is_alpha = !is_alpha`. I imagine this will get optimized by any decent compiler to have the same efficiency as a bitflip, and even if it isn't, this is going to be negligible.